sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -236,7 +236,8 @@ class SYCLMemObjT : public SYCLMemObjI {
     if (useHostPtr())
       throw invalid_parameter_error(
           "Buffer constructor from a pair of iterator values does not support "
-          "use_host_ptr property.");
+          "use_host_ptr property.",
+          PI_INVALID_OPERATION);
```

Comment:
Do you think PI_INVALID_VALUE is not apt for a invalid_parameter_error.
You have used that in the above file pretty consistently.

Reply:
There is no requirement in spec on exception in this case.
Though, for sake of consistency I'd rather change the type of exception to a mere `cl::sycl::runtime_error`.
Done.